include/http/middleware/system.js

```diff
@@ -18,16 +18,20 @@ module.exports = pb => ({
         req.handler.hostname = req.headers.host || pb.SiteService.getGlobalSiteContext().hostname;
         const siteObj = pb.RequestHandler.sites[req.handler.hostname];
 
-        if (siteObj) {
-            req.handler.siteObj = req.siteObj = siteObj;
+        if (siteObj && siteObj.prefix) {
```

Comment:
I just realized this solution could be much more simple. We can just replace the request.url. then it could solve all the routes, no matter static files or not. 

I kept the other codes as commented, you can check the difference here.

Reply:
I think we started making changes where it was more clear to us, without having deep knowledge of this part of the code. But since some time has passed I think this is only natural (this refactor)